### PR TITLE
NPE is ASTNode.squeezeSpaces method

### LINKED ISSUE #460: NPE is ASTNode.squeezeSpaces method
ktlint 0.37.1, diktat 0.1.2
```kotlin
/*
 * Copyright (c) Your Company Name Here. 2010-2020
 */

class HttpClient(var name: String) {
               var url: String = ""
    var port: String = ""
    var timeout = 0

    fun doRequest() {}
}

fun main() {
    val httpClient = HttpClient("myConnection")
        .apply {
            url = "http://pushkin.com"
            port = "8080"
            timeout = 100
        }
        .doRequest()
}
```
probably due to missing `package` directive

### PR DESCRIPTION
**Pull request by kentr0w**
### What's done:
   Fixed bug
   closes #460

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/CommentsFormatting.kt [hunk 1]
```diff
@@ -204,7 +204,7 @@ class CommentsFormatting(private val configRules: List<RulesConfig>) : Rule("kdo
     }
 
     private fun checkSpaceBetweenPropertyAndComment(node: ASTNode, configuration: CommentsFormattingConfiguration) {
-        if (node.treeParent.firstChildNode != node) {
+        if (node.treeParent.firstChildNode != node && node.treeParent.elementType != FILE) {
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/CommentsFormatting.kt:207]: Seems a little *костыльно*, maybe we can have separate branch (or just filtering) for top-level comments in the top method? It doesn't look well that this method is called `check space between property` and actually node can be top-level.

**Review by petertrr — COMMENTED** [review]
So, the issue was with top-level block comments? Can it happen with other types of comments?

**Comment by kentr0w** [comment]
No, I tested with `EOL_COMMENT` and `KDOC` and now there are no problems

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/CommentsFormatting.kt [hunk 2]
```diff
@@ -106,16 +109,16 @@ class CommentsFormatting(private val configRules: List<RulesConfig>) : Rule("kdo
...
             checkClassComment(node)
         }
-        checkWhiteSpaceBeforeComment(node, configuration)
+        checkWhiteSpaceBeginInComment(node, configuration)
     }
 
     private fun handleEolAndBlockComments(node: ASTNode, configuration: CommentsFormattingConfiguration) {
-        basicCommentsChecks(node, configuration)
-        checkWhiteSpaceBeforeComment(node, configuration)
+        if (node.treeParent.elementType != FILE)  basicCommentsChecks(node, configuration)
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/CommentsFormatting.kt:116]: What will happen if there is a top-level property with comment after it? It won't be checked? Like
```kotlin
package org.cqfn.diktat

val foo = 42// comment that has invalid spacing
```

Comment (kentr0w) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/CommentsFormatting.kt]: No, in this case comment belongs to property. But you are right about top-level, I fixed it.

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/CommentsFormatting.kt [hunk 3]
```diff
@@ -207,20 +200,27 @@ class CommentsFormatting(private val configRules: List<RulesConfig>) : Rule("kdo
...
-            } else if (!node.treePrev.textContains('\n') && node.treePrev.text.length != configuration.maxSpacesBeforeComment) {
-                // if there are too many spaces before comment
+        if (node.treeParent.firstChildNode == node)
+            return
+        if (node.treeParent.elementType == FILE) {
+            if (node.treePrev.isWhiteSpace()) {
                 COMMENT_WHITE_SPACE.warnAndFix(configRules, emitWarn, isFixMode,
-                        "There should be ${configuration.maxSpacesBeforeComment} space(s) before comment text, but there are too many in ${node.text}", node.startOffset, node) {
-                    (node.treePrev as LeafPsiElement).replaceWithText(" ".repeat(configuration.maxSpacesBeforeComment))
+                        "There should be 0 space(s) before comment text, but are none in ${node.text}", node.startOffset, node) {
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/CommentsFormatting.kt:208]: Not none, but `node.treePrev.text.count { it == ' ' }` ?

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/CommentsFormatting.kt [hunk 4]
```diff
@@ -207,20 +200,27 @@ class CommentsFormatting(private val configRules: List<RulesConfig>) : Rule("kdo
...
-                COMMENT_WHITE_SPACE.warnAndFix(configRules, emitWarn, isFixMode,
-                        "There should be ${configuration.maxSpacesBeforeComment} space(s) before comment text, but are none in ${node.text}", node.startOffset, node) {
-                    node.treeParent.addChild(PsiWhiteSpaceImpl(" ".repeat(configuration.maxSpacesBeforeComment)), node)
-                }
-            } else if (!node.treePrev.textContains('\n') && node.treePrev.text.length != configuration.maxSpacesBeforeComment) {
-                // if there are too many spaces before comment
+        if (node.treeParent.firstChildNode == node)
+            return
+        if (node.treeParent.elementType == FILE) {
+            if (node.treePrev.isWhiteSpace()) {
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/kdoc/CommentsFormatting.kt:206]: Also please add comment here, that this case is for top-level comments that are located in the beginning of the line and therefore don't need any spaces before.

**Review by petertrr — CHANGES_REQUESTED** [review]

**Review by petertrr — APPROVED** [review]

**Pull request merged**